commit cff130bf33f85cc3ab24f6584feaa227048c0738
Author: Kulikov Vasiliy <segooon@gmail.com>
Date:   Sun Oct 31 07:10:32 2010 +0000

    net: tipc: fix information leak to userland
    
    Structure sockaddr_tipc is copied to userland with padding bytes after
    "id" field in union field "name" unitialized.  It leads to leaking of
    contents of kernel stack memory.  We have to initialize them to zero.
    
    Signed-off-by: Vasiliy Kulikov <segooon@gmail.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

diff --git a/net/tipc/socket.c b/net/tipc/socket.c
index 230f9ca..296e28a 100644
--- a/net/tipc/socket.c
+++ b/net/tipc/socket.c
@@ -390,6 +390,7 @@ static int get_name(struct socket *sock, struct sockaddr *uaddr,
 	u32 portref = tipc_sk_port(sock->sk)->ref;
 	u32 res;
 
+	memset(addr, 0, sizeof(*addr));
 	if (peer) {
 		res = tipc_peer(portref, &addr->addr.id);
 		if (res)
